### ISSUE TITLE
💡 [FEAT] Taskwarrior integration

### ISSUE THREAD

**Issue by Saterlite** [kraanzu/dooit#38]
Is taskwarrior integration planned?

**Comment by kraanzu** [comment]
Like how exactly? Could you please explain more?

**Comment by cocoonkid** [comment]
I'd assume Dooit is more of a separate product. What specifically do you like to have implemented?

My opinion is that Dooit is a lot more lightweight than TW and that's why I like it :)

**Comment by ReneFroger** [comment]
But it would be possible to read the pending tasks from Taskwarrior? It's all stored in plain text anyway, so I don't see any problem with it.

**Comment by cocoonkid** [comment]
True. It might be possible to piggyback on taskd server integration?

That would be so cool :)

**Comment by kraanzu** [comment]
> But it would be possible to import or read the pending tasks from Taskwarrior? It's all stored in plain text anyway, so I don't see any problem with it.

That is true, but as cocoonkid pointed out:
> Dooit is more of a separate product

The two applications have very different approaches and it would only cause complications syncing the tasks.
I'll this close for now but, if in future, i changed my mind and decided to do it anyway, I'll re-open this thread :)

**Comment by kraanzu** [comment]
Hey @Saterlite and @ReneFroger,

> But it would be possible to import or read the pending tasks from Taskwarrior? It's all stored in plain text anyway, so I don't see any problem with it.

That is true, but the current implementation was unsuitable for this change.
I am working on a new release that will introduce a bunch of new features :smiley: and would make the implementation much much better 

And I have this idea in mind for now:
There will be a plugin system with let's say two functions namely
- entry code 
- exit code

The entry code will parse taskwarrior or any other todo manager's todos and connect it to Dooit creating a new workspace 
The exit code will then sync any changes made to the original app's Todos :)

What are your thoughts on this?

**Comment by figjeti** [comment]
There is also called a project called taskwarrior-tui and it is pretty fully feature, just to point you in another possible direction if it suites you better.
